plugin_tests/web_client_test.py

```diff
@@ -12,9 +15,77 @@
 )
 
 
+class MockSlicerCLIWebResource(Resource):
+    """
+    This creates a mocked version of the ``/HistomicsTK/HistomicsTK/docker_image``
+    endpoint so we can test generation of the analysis panel on the client without
+    relying on girder_worker + docker.
+    """
+
+    def __init__(self):
+        super(MockSlicerCLIWebResource, self).__init__()
+        self.route('GET', ('docker_image',), self.dockerImage)
+        self.route('GET', ('test_analysis', 'xmlspec'), self.testAnalysisXml)
+        self.route('POST', ('test_analysis', 'run'), self.testAnalysisRun)
+
+    @access.public
+    @describeRoute(
+        Description('Mock the /HistomicsTK/docker_image endpoint.')
+    )
+    def dockerImage(self, params):
+        """
+        Return a single CLI referencing mocked out /xmlspec and /run endpoints.
+        """
+        return {
+            'dsarchive/histomicstk': {
+                'latest': {
+                    'NucleiDetection': {
+                        'run': 'mock_resource/test_analysis/run',
+                        'type': 'python',
+                        'xmlspec': 'mock_resource/test_analysis/xmlspec'
+                    }
+                }
+            }
+        }
+
+    @access.public
+    @describeRoute(
+        Description('Mock an analysis description route.')
+    )
+    def testAnalysisXml(self, params):
+        """Return the nuclei detection XML spec as a test case."""
+        xml_file = os.path.join(
+            os.path.abspath(
+                os.path.join(
+                    os.path.dirname(__file__), 'test_files', 'NucleiDetection.xml'
```

Comment:
Can we change this to `os.path.dirname(__file__), '..', 'server', 'NucleiDetection', 'NucleiDetection.xml'` and not duplicate the NucleiDetection.xml file?